### ISSUE TITLE
glob pattern descends into nested workspace

### ISSUE THREAD

**Issue by alexeagle** [bazelbuild/bazel#4242]
bazel 0.8.1

`WORKSPACE`
```
workspace(name = "parent")

local_repository(
  name = "child",
  path = "./child",
)
```

`BUILD.bazel`
```
filegroup(name = "parent", srcs = glob(["**/*.txt"]))
```

`child/WORKSPACE`
```
workspace(name = "child")
```

`child/BUILD.bazel` [empty]
`child/foo.txt` [empty]

```
bazel build ...
ERROR: /usr/local/google/home/alexeagle/repro_bazel_nested_workspace/BUILD.bazel:1:35: Label '//:child/foo.txt' crosses boundary of subpackage '@child//' (have you deleted @child///BUILD? If so, use the --deleted_packages=@child// option)
ERROR: package contains errors: 
ERROR: error loading package '': Package '' contains errors
```

**Comment by katre** [comment]
Thank you for reporting these. I'll try and figure out the issue behind it.

**Comment by katre** [comment]
Actually, I just tested this with Bazel HEAD, and it works correctly. I think the fix will be in 0.9.0, I'll check later on today and report back.

**Comment by katre** [comment]
This was fixed by the combination of 4117c867fe8e560f53bc1c7106af9c2889cc18f2 and e0d1d0ef7e1d53f232a457ac66496a5d8d9b82e3 (one fixes GlobFunction, the other fixes PathPackageLocator to consider BUILD.bazel).

Based on the dates, I don't think either is in 0.8.1, but both should be in 0.9.0, so once that release is out this should be dealt with.

Thanks!

**Comment by alexeagle** [comment]
Okay thanks
